File: iosApp/iosApp/Model/GameChoice.swift
//
//  GameChoice.swift
//  CACDGAME
//
//  Created by François Dabonot on 03/07/2023.
//  Copyright © 2023 CACD2. All rights reserved.
//

import Foundation
import Shared
import SwiftUI

extension GameChoice {

    func image(selectedGameId: GameChoice?) -> String {
        let isSelected = selectedGameId?.gameId == gameId
        switch self {
        case .design:
            if isSelected {
                return R.image.design_selected.name
            } else {
                return R.image.design_normal.name
            }
        case .product:
            if isSelected {
                return R.image.product_selected.name
            } else {
                return R.image.product_img.name
            }
        case .tech:
            if isSelected {
                return R.image.tech_selected.name
            } else {
                return R.image.tech_normal.name
            }
        default:
            if isSelected {
                return R.image.mix_selected.name
            } else {
                return R.image.mix_normal.name
            }
        }
    }

    func title() -> LocalizedStringKey {
        switch self {
        case .design:
            return R.string.localizable.selection_quiz_design.localizedKey
        case .product:
            return R.string.localizable.selection_quiz_produit.localizedKey
        case .tech:
            return R.string.localizable.selection_quiz_tech.localizedKey
        default:
            return R.string.localizable.selection_mix.localizedKey
        }
    }

    func label() -> LocalizedStringKey {
        switch self {
        case .design:
            return R.string.localizable.label_design.localizedKey
        case .product:
            return R.string.localizable.label_produit.localizedKey
        case .tech:
            return R.string.localizable.label_tech.localizedKey
        default:
            return R.string.localizable.label_mix.localizedKey
        }
    }

    func color(selectedGameId: GameChoice?) -> Color {
        let isSelected = selectedGameId?.gameId == self.gameId
        if isSelected {
            return R.color.primary.color
        } else {
            return  R.color.onSurface.color
        }
    }

}

File: androidApp/src/main/java/com/cacd2/cacdgame/android/model/GameChoice.kt
package com.cacd2.cacdgame.android.model

import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import androidx.compose.material.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import com.cacd2.cacdgame.android.R
import com.cacd2.cacdgame.model.GameChoice

/**
 * Created by frankois on 15/05/2023.
 */
@DrawableRes
fun GameChoice.image(selectedGameId: GameChoice?): Int {
    val isSelected = selectedGameId?.gameId == this.gameId
    return when (this) {
        GameChoice.DESIGN -> {
            if (isSelected) {
                R.drawable.design_selected
            } else {
                R.drawable.design_interface_img
            }
        }

        GameChoice.PRODUCT -> {
            if (isSelected) {
                R.drawable.product_selected
            } else {
                R.drawable.product_img
            }
        }

        GameChoice.TECH -> {
            if (isSelected) {
                R.drawable.tech_selected
            } else {
                R.drawable.tech_img
            }
        }

        GameChoice.ALL -> {
            if (isSelected) {
                R.drawable.mix_selected
            } else {
                R.drawable.mix
            }
        }
    }
}

val GameChoice.title: Int
    @StringRes
    get() =
        when (this) {
            GameChoice.DESIGN -> {
                R.string.selection_quiz_design
            }

            GameChoice.PRODUCT -> {
                R.string.selection_quiz_produit
            }

            GameChoice.TECH -> {
                R.string.selection_quiz_tech
            }

            GameChoice.ALL -> {
                R.string.selection_mix
            }
        }

val GameChoice.label: Int
    @StringRes
    get() =
        when (this) {
            GameChoice.DESIGN -> {
                R.string.label_design
            }

            GameChoice.PRODUCT -> {
                R.string.label_produit
            }

            GameChoice.TECH -> {
                R.string.label_tech
            }

            GameChoice.ALL -> {
                R.string.label_mix
            }
        }

@Composable
fun GameChoice.color(selectedGameId: GameChoice?): Color {
    val isSelected = selectedGameId?.gameId == this.gameId
    return if (isSelected) {
        MaterialTheme.colors.primary
    } else {
        MaterialTheme.colors.onSurface
    }
}
